File: backend/app/schemas/brand.py
from pydantic import BaseModel, HttpUrl
from typing import Optional

class BrandBase(BaseModel):
    name: str
    
    
class BrandCreate(BrandBase):
    logo_url: Optional[HttpUrl] = None
    

class BrandUpdate(BaseModel):
    name: Optional[str] = None
    logo_url: Optional[str] = None


class BrandOut(BrandBase):
    id: int
    logo_url: Optional[HttpUrl] = None
    
    class Config:
        from_attributes = True

File: frontend/src/types/brand.ts
export interface BrandOut {
    id: number;
    name: string;
    logo_url: string | null;
}

export interface BrandCreate {
    name: string;
    logo_url?: string | null;
}

export interface BrandUpdate {
    name?: string | null;
    logo_url?: string | null;
}
